turbojpeg.py

```diff
@@ -260,7 +291,63 @@ def scale_with_quality(self, jpeg_buf, scaling_factor=None, quality=85, flags=0)
         finally:
             self.__destroy(handle)
 
+    def crop(self, jpeg_buf, x, y, w, h, gray=False):
+        """losslessly crop a jpeg image with optional grayscale"""
+        handle = self.__init_transform()
+        try:
+            jpeg_array = np.frombuffer(jpeg_buf, dtype=np.uint8)
+            src_addr = self.__getaddr(jpeg_array)
+            width = c_int()
+            height = c_int()
+            jpeg_colorspace = c_int()
+            jpeg_subsample = c_int()
+            status = self.__decompress_header(
+                handle, src_addr, jpeg_array.size, byref(width), byref(height),
+                byref(jpeg_subsample), byref(jpeg_colorspace))
+            if status != 0:
+                self.__report_error(handle)
+            if x < 0:
+                x = 0
+            else:
+                x = (x / 16) * 16
```

Comment:
16x16 will work for the images we have (so would 16x8).  But I would rather not hard code it.  Look up the MCU width and height by the `jpeg_subsample` value.  https://cdn.rawgit.com/libjpeg-turbo/libjpeg-turbo/master/doc/html/group___turbo_j_p_e_g.html#gabd247bb9fecb393eca57366feb8327bf

Reply:
Also...if we adjust the x or y should we adjust w and h by the same amount so that the crop ends on the same pixel?